script/tool/lib/src/branch_for_batch_release_command.dart

```diff
@@ -0,0 +1,327 @@
+// Copyright 2013 The Flutter Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' as io;
+import 'dart:math' as math;
+
+import 'package:file/file.dart';
+import 'package:git/git.dart';
+import 'package:pub_semver/pub_semver.dart';
+import 'package:yaml/yaml.dart';
+import 'package:yaml_edit/yaml_edit.dart';
+
+import 'common/core.dart';
+import 'common/output_utils.dart';
+import 'common/package_command.dart';
+import 'common/repository_package.dart';
+
+const int _kExitPackageMalformed = 3;
+const int _kGitFailedToPush = 4;
+
+// The template file name used to draft a pending changelog file.
+// This file will not be picked up by the batch release process.
+const String _kTemplateFileName = 'template.yaml';
+
+/// A command to create a remote branch with release changes for a single package.
+class BranchForBatchReleaseCommand extends PackageCommand {
+  /// Creates a new `branch-for-batch-release` command.
+  BranchForBatchReleaseCommand(
+    super.packagesDir, {
+    super.processRunner,
+    super.platform,
+    super.gitDir,
+  }) {
+    argParser.addOption(
+      'branch',
+      mandatory: true,
+      abbr: 'b',
+      help: 'The branch name to contain the release commit',
+    );
+    argParser.addOption(
+      'remote',
+      mandatory: true,
+      abbr: 'r',
+      help: 'The remote to push the branch to.',
+    );
+  }
+
+  @override
+  final String name = 'branch-for-batch-release';
+
+  @override
+  final String description = 'Creates a release PR for a single package.';
+
+  @override
+  Future<void> run() async {
+    final String branchName = getStringArg('branch');
+    final String remoteName = getStringArg('remote');
+
+    final List<RepositoryPackage> packages = await getTargetPackages()
+        .map((PackageEnumerationEntry e) => e.package)
+        .toList();
+    if (packages.length != 1) {
+      printError('Exactly one package must be specified.');
+      throw ToolExit(2);
+    }
+    final RepositoryPackage package = packages.single;
+
+    final GitDir repository = await gitDir;
+
+    print('Parsing package "${package.displayName}"...');
+    final _PendingChangelogs pendingChangelogs =
+        await _getPendingChangelogs(package);
+    if (pendingChangelogs.entries.isEmpty) {
+      print('No pending changelogs found for ${package.displayName}.');
+      return;
+    }
+
+    final Pubspec pubspec =
+        Pubspec.parse(package.pubspecFile.readAsStringSync());
+    if (pubspec.version == null || pubspec.version!.major < 1) {
+      printError(
+          'This script only supports packages with version >= 1.0.0. Current version: ${pubspec.version}. Package: ${package.displayName}.');
```

Comment:
The package name has already been logged above, so doesn't need to be repeated here.